locksmith/src/utils/middlewares/eventOrganizerMiddleware.ts

```diff
@@ -1,63 +1,30 @@
 import { RequestHandler } from 'express'
-import { Web3Service } from '@unlock-protocol/unlock-js'
-import { PaywallLocksConfigType } from '@unlock-protocol/core'
-import networks from '@unlock-protocol/networks'
-import { getEventBySlug } from '../../operations/eventOperations'
-import { getCheckoutConfigById } from '../../operations/checkoutConfigOperations'
-import { EventBody } from '../../controllers/v2/eventsController'
-import { isEmpty } from 'lodash'
+import { isEventOrganizer } from '../eventOrganizers'
 
 export const eventOrganizerMiddleware: RequestHandler = async (
   request,
   response,
   next
 ) => {
-  const web3Service = new Web3Service(networks)
   const userAddress = request.user!.walletAddress
 
   let slug = request.params.slug
-  let locks: PaywallLocksConfigType = {}
 
   if (!slug) {
-    const parsed = await EventBody.parseAsync(request.body)
-
-    locks = parsed.checkoutConfig.config.locks
-    slug = parsed.data.slug
+    return response.status(404).send({
+      message: `No such event`,
+    })
   }
 
-  // If this is an existing event!
-  if (slug) {
-    const existingEvent = await getEventBySlug(
-      slug,
-      false /** includeProtected */
-    )
-    if (existingEvent?.checkoutConfigId) {
-      const checkoutConfig = await getCheckoutConfigById(
-        existingEvent.checkoutConfigId
-      )
-      locks = checkoutConfig?.config.locks || {}
-    }
-  }
+  const isLockManager = await isEventOrganizer(userAddress, slug)
 
-  if (isEmpty(locks)) {
+  if (isLockManager == undefined) {
```

Comment:
That does not seem right... We should either check if the event exists or just use `if (isLockManager == false) {` below. We should not conflate 2 unrelated things (event exists and user is lock manager)